src/util-napatech.c

```diff
@@ -0,0 +1,755 @@
+/* Copyright (C) 2017 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+/**
+ * \file
+ *
+ * \author Napatech Inc.
+ * \author Phil Young <py@napatech.com>
+ *
+ * 
+  */
+
+
+#include "suricata-common.h"
+#ifdef HAVE_NAPATECH
+#include "suricata.h"
+#include "threadvars.h"
+#include "tm-threads.h"
+
+
+uint16_t NapatechGetNumaNode(uint16_t stream_id) 
+{
+    int status;
+    char buffer[80]; // Error buffer
+    NtInfoStream_t info_stream;
+    NtInfo_t info;
+    uint16_t numaNode;
+
+    if ((status = NT_InfoOpen(&info_stream, "SuricataStreamInfo")) != NT_SUCCESS) {
+        NT_ExplainError(status, buffer, sizeof (buffer) - 1);
+        SCLogError(SC_ERR_NAPATECH_STREAMS_REGISTER_FAILED, "NT_InfoOpen failed: %s", buffer);
+        exit(EXIT_FAILURE);
+    }
+
+    // Read the info on this specific stream
+    info.cmd = NT_INFO_CMD_READ_STREAMID;
+    info.u.streamID.streamId = stream_id;
+    if ((status = NT_InfoRead(info_stream, &info)) != NT_SUCCESS) {
+        NT_ExplainError(status, buffer, sizeof (buffer) - 1);
+        SCLogError(SC_ERR_NAPATECH_INIT_FAILED, "NT_InfoRead() failed: %s\n", buffer);
+        exit(EXIT_FAILURE);
+    }
+
+    numaNode = info.u.streamID.data.numaNode;
+
+    if ((status = NT_InfoClose(info_stream)) != NT_SUCCESS) {
+        NT_ExplainError(status, buffer, sizeof (buffer) - 1);
+        SCLogError(SC_ERR_NAPATECH_STREAMS_REGISTER_FAILED, "NT_InfoClose failed: %s", buffer);
+        exit(EXIT_FAILURE);
+    }
+    return numaNode;
+}
+
+
+//-----------------------------------------------------------------------------
+//------------------------------------------------------------------------------
+// Statistics code
+//-----------------------------------------------------------------------------
+
+typedef struct StreamCounters_ {
+    uint16_t pkts;
+    uint16_t byte;
+    uint16_t drop;
+} StreamCounters;
+
+
+NapatechCurrentStats current_stats[MAX_STREAMS];
+
+NapatechCurrentStats NapatechGetCurrentStats(uint16_t id) 
+{
+
+    return current_stats[id];
+}
+
+
+static uint16_t TestStreamConfig(
+        NtInfoStream_t hInfo,
+        NtStatStream_t hStatStream,
+        NapatechStreamConfig stream_config[], 
+        uint16_t num_inst) 
+{
+    int status;
+    char buffer[80]; // Error buffer
+    NtStatistics_t stat; // Stat handle.
```

Comment:
if this gets reset in each loop iteration I prefer it to be declared in the loop